best-practices/native/nativescript/evironment.md

```diff
@@ -0,0 +1,80 @@
+The best practice recommends not to use more than 2 properties, `production` and `apiUrl`.
+To achieve that, we need to make some changes on environment files as you can see below:
+```
+https://medium.com/most-wanted-experts/angular-application-configurations-best-practices-2e28514b9b4e
+```
+
+The configuration on `angular.json` does not make effect when we buld the application in NativeScript. Below we have a work around to use environment.
+```
+https://medium.com/@derekfong/nativescript-angular-setup-environment-ts-for-different-environments-631b4c5219b6
+```
+
+1 - This goes to `environment.<dev|stage|prod>.ts`
+```typescript
+export const environment = {
+    production: true,
+    apiUrl: 'http://www.yoursite.com'
+};
+```
+
+2 - On environment.tns.ts:
+```typescript
+import { environment as devEnvironment } from './environment.dev';
+import { environment as prodEnvironment } from './environment.prod';
+// TODO: import additional environment (e.g. uat) if required.
+export const environment = (() ={
+  let envVars;
+  if (Object.prototype.hasOwnProperty.call(process.env, 'environment') &&
+        process.env.environment
+  ) {
+    switch (process.env.environment) {
+      case 'prod':
+        envVars = prodEnvironment;
+        break;
+      // TODO: Add additional environment (e.g. stage) if required. 
+      default:
+        envVars = devEnvironment;
+    }
+  } else {
+    envVars = devEnvironment;
+  }
+  return envVars;
+})();
+```
+
+Also we need make some changes on "webpack.config.js".
+So, look for "new webpack.DefinePlugin" on "webpack.config.js". in that place modify to looks like this:
+ 
+```typescript
+new webpack.DefinePlugin({
+        'global.TNS_WEBPACK': 'true',
+        // 'process': undefined,
```

Comment:
Why do we want to leave this commented out code here?

Reply:
I just copy from Lint sample. I will removed.